packages/manager/apps/pci-private-network/src/data/hooks/networks/useNetworks.ts

```diff
@@ -53,3 +148,38 @@ export const useSubnets = (
     queryFn: () => getSubnets(projectId, region, networkId),
     enabled: !!(projectId && networkId && region),
   });
+
+export const addPrivateNetwork = async (
+  projectId: string,
+  newNetwork: TNetwork,
+) => {
+  const queryKey = networkQueryKey(projectId);
+  const networks = queryClient.getQueryData<TNetwork[]>(queryKey);
+
+  if (!networks) {
+    await fetchPrivateNetwork(projectId);
```

Comment:
I don't understand why there is this call. Same below.

Reply:
it is an optimistic add. If `getQueryData` return undefined for some reason I refetch from api and update the cache. Otherwise I juste update the data cache

Reply:
Why not just invalidate the query ?